src/ConnectedMode/UI/ManageConnections/ManageConnectionsDialog.xaml.cs

```diff
@@ -52,11 +52,18 @@ private void EditConnection_Clicked(object sender, RoutedEventArgs e)
 
         private void NewConnection_Clicked(object sender, RoutedEventArgs e)
         {
-            if (GetTransientConnection() is {} partialConnection && GetCredentialsDialog(partialConnection) is {} credentialsDialog && 
-                CredentialsDialogSucceeded(credentialsDialog) && GetCompleteConnection(partialConnection, credentialsDialog) is {} completeConnection)
+            if (GetTransientConnection() is not {} transientConnection)
             {
-               ViewModel.AddConnection(new Connection(completeConnection));
+                return;
+            }
+
+            var credentialsDialog = GetCredentialsDialog(transientConnection);
+            if (!CredentialsDialogSucceeded(credentialsDialog) || GetCompleteConnection(transientConnection, credentialsDialog) is not { } completeConnection)
```

Comment:
at this point I would extract the server type check from GetCompleteConnection and rename it to FinalizeSonarCloudConnection or smth like that

Reply:
I would rather leave it like this. I think it is more readable and less cognitive complexity. It is, in the end, related to the final connection.

Would you want me to rename the GetCompleteConnection to FinalizeConnection? I see that SlCore also has the Transient connection vs full connection. So for me seems fine to stay as it is.

Reply:
I renamed the method

Reply:
1) it hides the fact that this method is only really needed to be called for SC connections, which just makes it less clear why we call it for SQ
2) both of these connections are transitive, as they need to be saved in the repository first and then synced with SLCore to be non-transitive

Reply:
I don't think the caller needs to know that is only needed for SC. It simply needs the final connection. Adding another (a || (b&&c) ) makes the if more complex, therefore I think the logic should be part of another method